File: test-suite-python/src/test/python/micronaut/docs/metrics/annotation/MetricOptionsFilterTaggersExample.py
# tag::imports[]
from io.micrometer.core.annotation import Timed
from jakarta.inject import Singleton
from micronaut.configuration.metrics.annotation import MetricOptions

from .MethodNameTagger import MethodNameTagger
# end::imports[]


# tag::class[]
@Singleton
class MetricOptionsFilterTaggersExample:

    @MetricOptions(
        filterTaggers=True,  # Specify that not all taggers should be applied
        taggers=[MethodNameTagger]  # Specific taggers to apply
    )
    @Timed(value="do_something")
    def do_something(self) -> None:
        pass
# end::class[]

File: test-suite/src/test/java/io/micronaut/docs/metrics/annotation/MetricOptionsFilterTaggersExample.java
package io.micronaut.docs.metrics.annotation;

// tag::imports[]
import io.micrometer.core.annotation.Timed;
import io.micronaut.configuration.metrics.annotation.MetricOptions;
import jakarta.inject.Singleton;
// end::imports[]

// tag::class[]
@Singleton
public class MetricOptionsFilterTaggersExample {

    @MetricOptions(
        filterTaggers = true, // Specify that not all taggers should be applied
        taggers = {MethodNameTagger.class} // Specific taggers to apply
    )
    @Timed(value = "do_something")
    public void doSomething() {
        // ...
    }
}
// end::class[]
